test/unittests/OpModel/TTNN/Lib/TestOpModelLib.cpp

```diff
@@ -2636,6 +2636,116 @@ TEST_P(OpModelAvgPool2DParam, AvgPool2DParam) { RunTest(); }
 INSTANTIATE_TEST_SUITE_P(AvgPool2DTests, OpModelAvgPool2DParam,
                          pool2DTestValues);
 
+class OpModelGlobalAvgPool2dParam
+    : public OpModelTest,
+      public testing::WithParamInterface<
+          std::tuple<detail::TestTensor,    // input
+                     detail::TestTensor,    // output
+                     detail::ExpectedResult // expected result
+                     >> {
+protected:
+  void RunTest() {
+    auto params = GetParam();
+    const auto [inputShape, inputTensorLayout, inputBufferType,
+                inputVirtualGrid] = std::get<0>(params);
+    const auto [outputShape, outputTensorLayout, outputBufferType,
+                outputVirtualGrid] = std::get<1>(params);
+    const auto [expectedLegal, expectedCbSize, expectedL1PeakSize,
+                expectedTotalPeakSize, expectedOutputSize] =
+        std::get<2>(params);
+
+    const TTNNLayoutAttr inputLayout = CreateTiledLayout(
+        inputShape, inputBufferType, inputTensorLayout, inputVirtualGrid);
+    const TTNNLayoutAttr outputLayout = CreateTiledLayout(
+        outputShape, outputBufferType, outputTensorLayout, outputVirtualGrid);
+
+    auto constraintsExp = OpModel<GlobalAvgPool2dOp>::getOpConstraints(
+        CreateWorkerGrid(), inputShape, inputLayout, outputLayout);
+
+    EXPECT_EQ(static_cast<bool>(constraintsExp), expectedLegal);
+    if (expectedLegal) {
+      const auto [cbSize, l1PeakSize, totalPeakSize, outputSize,
+                  outputLayoutReadBack] = constraintsExp.get();
+      EXPECT_GT(cbSize, 0);
+      EXPECT_GT(l1PeakSize, 0);
+      EXPECT_GT(outputSize, 0);
+      ExpectLayoutsEQ(outputLayout, outputLayoutReadBack);
```

Comment:
Let's utilize `std::get<2>(params)` for assertions. Also, we need to assert `totalPeakSize` values:
```suggestion
      EXPECT_GT(cbSize, expectedCbSize);
      EXPECT_GT(l1PeakSize, expectedL1PeakSize);
      EXPECT_GT(totalPeakSize, expectedTotalPeakSize);
      EXPECT_GT(outputSize, expectedOutputSize);
      ExpectLayoutsEQ(outputLayout, outputLayoutReadBack);
```